network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -252,4 +269,30 @@ public String toString() {
         .toString();
     }
   }
+
+  /**
+   * write out the set of registered executors to a file so we can reload them on restart.
+   * You must have a lock on executors when calling this
+   */
+  private void saveRegisteredExecutors() throws IOException {
+    if (registeredExecutorFile != null) {
+      Options options = new Options();
+      options.createIfMissing(true);
+      JniDBFactory factory = new JniDBFactory();
+      DB db = null;
+      try {
+        db = factory.open(registeredExecutorFile, options);
```

Comment:
I wonder how expensive this is.  Any reason to not just open it once in constructor and have global db variable?